src/components/LibrariesPage.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+import { Store, Reducer } from "redux";
+import { State } from "../reducers/index";
+import { LibrariesData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import LibrariesList from "./LibrariesList";
+import SearchForm from "./SearchForm";
+
+export interface LibrariesPageStateProps {
+  libraries?: LibrariesData;
+  results?: LibrariesData;
+}
+
+export interface LibrariesPageOwnProps {
+  store: Store<State>;
+}
+
+export interface LibrariesPageDispatchProps {
+  fetchData: () => Promise<LibrariesData>;
+  search: (data: FormData) => Promise<LibrariesData>;
+}
+
+export interface LibrariesPageState {
+  showAll: boolean;
+  searchTerm: string;
+}
+
+export interface LibrariesPageProps extends LibrariesPageStateProps, LibrariesPageOwnProps, LibrariesPageDispatchProps {};
+
+export class LibrariesPage extends React.Component<LibrariesPageProps, LibrariesPageState> {
+  constructor(props: LibrariesPageProps) {
+    super(props);
+    this.search = this.search.bind(this);
+    this.clear = this.clear.bind(this);
+    this.state = { showAll: true, searchTerm: "" };
+  }
+
+  render(): JSX.Element {
+    let hasLibraries = (this.props.libraries && this.props.libraries.libraries && this.props.libraries.libraries.length > 0);
+    let allLibraries = hasLibraries && this.props.libraries.libraries;
+    let results = !this.state.showAll && this.props.results && this.props.results.libraries;
```

Comment:
This makes sense but reading it is difficult because it's a complicated set of logic statements. Can you write some comments? I would say that if someone else reading it may have trouble breaking it down easily, comments or more separate steps would be helpful.